### PR TITLE
Add support for java.time.Year as JAX-RS parameter

### LINKED ISSUE #34324: resteasy-reactive java.time.Year ParamConverter
### Description

Please consider adding a ParamConverter in resteasy-reactive for java.time.Year.

https://github.com/quarkusio/quarkus/tree/main/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Closes: #34324

### CODE REVIEW

**Review by Sgitario — APPROVED** [review]
lgtm!

**Comment by gastaldi** [comment]
The test error seems related

File: independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    @Test
+    public void localDateAsQueryParam() {
+        RestAssured.get("/hello?date=01984")
+                .then().statusCode(200).body(Matchers.equalTo("hello#1984"));
+    }
+
+    @Test
+    public void localDateAsPathParam() {
+        RestAssured.get("/hello/-21")
+                .then().statusCode(200).body(Matchers.equalTo("hello@-21"));
```
Comment (bulldog98) [inline independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: I also do not see how -21 is a valid year

Comment (geoand) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: It is

**Comment by geoand** [comment]
Yeah, it's weird that it only happens on one run and also not locally for me.

But I'll have a look tomorrow

File: independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java (hunk 1 above)
Comment (bulldog98) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: ok now I'm suprised by the time api.

Comment (bulldog98) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: if I understood this correctly "-21" is passed to `YearParamConverter.convert`, and the parser fails with that input.

Comment (geoand) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: Try `Year.parse("-21")` in jshell and all your doubts will be dispelled

Comment (bulldog98) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java:39]: ok I tried it with `PARSER`, my bad

**Pull request merged**